haystack/document_stores/opensearch.py

```diff
@@ -241,20 +241,14 @@ def _init_client(
         if use_system_proxy:
             connection_class = RequestsHttpConnection
 
-        if username:
-            # standard http_auth
-            client = OpenSearch(
-                hosts=hosts,
-                http_auth=(username, password),
-                scheme=scheme,
-                ca_certs=ca_certs,
-                verify_certs=verify_certs,
-                timeout=timeout,
-                connection_class=connection_class,
-            )
-        elif aws4auth:
+        if aws4auth:
             # Sign requests to Opensearch with IAM credentials
             # see https://docs.aws.amazon.com/opensearch-service/latest/developerguide/request-signing.html#request-signing-python
+            if username and username != "admin":
+                logger.warning(
```

Comment:
pylint fails for this in the CI with 
`haystack/document_stores/opensearch.py:248:16: W1201: Use lazy % or .format() formatting in logging functions (logging-not-lazy)`
I think it's because of the string concatenation with `+`. Please put the string in one line. An alternative is to use fstrings like the following:
```python
logger.warning(
                f"aws4auth and a username are passed to the OpenSearchDocumentStore. "
                f"The username will be ignored and aws4auth will be used for authentication."
)
```
Right now, no matter whether the log level is set to warning or not, the concatenation operation will always be executed. It's a little bit of extra work that is not needed if the level is not set to warning and that's causing the `logging-not-lazy` complaint.

Reply:
Interesting! I did not know that this slows down logging. I changed it to a really long string, hope that is ok.